src/qililab/result/stream_results.py

```diff
@@ -0,0 +1,212 @@
+# Copyright 2023 Qilimanjaro Quantum Tech
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Any
+
+import h5py
+import numpy as np
+
+
+def stream_results(shape: tuple, path: str, loops: dict[str, np.ndarray]):
+    """Constructs a StreamArray instance.
+
+    This methods serves as a constructor for user interface of the StreamArray class.
+
+    Args:
+        shape (tuple): results array shape.
+        path (str): path to save results.
+        loops (dict[str, np.ndarray]): dictionary with each loop name in the experiment as key and numpy array as values.
+
+    Examples:
+
+        .. note::
+
+            All the following examples are explained in detail in the :ref:`Platform <platform>` section of the documentation. However, here are a few thing to keep in mind:
```

Comment:
Juanjo did you copy this from somewhere else? 😅 

Let's try to keep the examples short and concise please! Otherwise nobody will read them 🙏 

I don't think you should be talking about the Platform here... you should only add an example of how this function works. If you want (and if needed), you can always add a **small** note that links to the platform documentation.

So please, remove everything in this example that is not related to the `stream_results` function 🙏

Reply:
Also, can you add an extra example that explains how to use this with the `%%submit_job` magic method of SLURM?